### PR TITLE
Fix Python S3 API

### PR DESCRIPTION
**Pull request by bradyoo**

### CODE REVIEW

**Comment by bradyoo** [comment]
I mistakenly thought that the accesskey would be empty if user didn't provide anything. In reality, it was something like AWS :<Secret>

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java [hunk 1]
```diff
@@ -125,7 +125,12 @@ private static String getUserFromAuthorization(String authorization) {
       return null;
     }
 
-    return stripped.substring(0, colonIndex);
+    final String user = stripped.substring(0, colonIndex);
```
Comment (gpang) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:128]: is there a unittest we can add to verify this behavior?

**Review by gpang — COMMENTED** [review]
@bradyoo

**Comment by bradyoo** [comment]
Unit test added.

**Review by gpang — APPROVED** [review]
LGTM

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**